SIGNALESP/SIGNALESP.ino

```diff
@@ -247,16 +248,41 @@ void loop() {
 		if (!command_available) { cmdstring = ""; }
 		blinkLED = true;
 	}
+	
 	if (fifousage < FiFo.count())
 	  fifousage = FiFo.count();
-
-	while (FiFo.count()>0) { //Puffer auslesen und an Dekoder uebergeben
-		aktVal = FiFo.dequeue();
-		state = musterDec.decode(&aktVal);
-		if (state) blinkLED = true; //LED blinken, wenn Meldung dekodiert
-		if (FiFo.count()<120) yield();
+  while (FiFo.count()>0) { //Puffer auslesen und an Dekoder uebergeben
+  		aktVal = FiFo.dequeue();
+	  	state = musterDec.decode(&aktVal);
+		  if (state) blinkLED = true; //LED blinken, wenn Meldung dekodiert
+//      if (FiFo.count()<120) yield();
```

Comment:
Könntest Du diesen Eintrag bitte aktivieren. Die while schleife könne durchaus längere Zeit laufen, wenn viele Einträge im FIFO sind.

Reply:
gerade bei einer vollen Queue würde yield() nicht mehr aufgerufen werden (< 120). Bei wenigen Einträgen dafür ständig.

Reply:
Das war die Idee, damit wir den Puffer wieder leeren und nicht viel Zeit mit yield() verbringen, wenn der puffer gut gefüllt ist.

Reply:
Wenn die Schleife sehr viel Zeit benötigt, dann soll durch yield() doch eine Pause der Schleife stattfinden, damit andere Aufgaben erledigt werden können. Somit würde ich erwarten, dass yield() bei mehr Einträgen in der Queue häufiger aufgerufen wird.

Reply:
Ich verstehe deinen Standpunkt.
Da ich nicht weiss, wie lange yield() benötigt, war mein Ansatz eher Mal Platz im Puffer zu schaffen, bevor ein yield aufgerufen wird.
Der Puffer muss ja in der Regel schneller entleert werden, als das neue Einträge hinzu kommen. Wenn nun yield bei jedem Aufruf 100 uS beträgt, wäre das schon grenzwertig. 
Daher dachte ich, erst Mal den Puffer etwas abarbeiten und dann yield aufrufen.
Alternativ könnte man auch darüber nachdenken, yield z.B. bei jedem 4.  Eintrag laufen zu lassen, dann ist das Verhalten gleichmäßig über den Füllstand verteilt.

Reply:
auch ein timeout für das letzte yield() wäre akzeptable (z.B. alle 20 ms). leider bekommen wir nicht mit, wann in signalDecoder Nachrichten versandt werden. 
Kannst du dir einen Callback vorstellen?